tests/unit/dataset/dataset_definition_test.py

```diff
@@ -365,3 +365,23 @@ def test_check_equality_of_load_from_yaml_and_load_from_dictionary_dump(tmp_path
     )
 
     assert yaml_definition == expected_definition
+
+
+def test_dataset_to_dict_exclude_none():
+    definition = DatasetDefinition(
+        name='Example',
+        trial_columns=[],
+        experiment=Experiment(origin=None),
+    )
+
+    dict_default = definition.to_dict()
+    assert 'long_name' not in dict_default
+    assert 'trial_columns' not in dict_default
+    assert 'has_files' not in dict_default
+    assert 'experiment' not in dict_default
+
+    dict_non_default = definition.to_dict(exclude_none=False)
+    assert 'long_name' in dict_non_default
+    assert 'trial_columns' in dict_non_default
+    assert 'has_files' in dict_non_default
+    assert 'experiment' in dict_non_default
```

Comment:
Unit tests should test a specific case. This function tests multiple things. It is recommended to write different test functions for different cases, or even better, use `@pytest.mark.parametrize` for different test parameters.

It shouldn't take long but improves readability and maintainability on the long run by a lot.